lib/streamy/message_buses/kafka_message_bus.rb

```diff
@@ -1,29 +1,23 @@
 require "streamy/kafka_configuration"
-require "kafka"
 require "active_support/core_ext/hash/indifferent_access"
 require "active_support/json"
 
 module Streamy
   module MessageBuses
     class KafkaMessageBus < MessageBus
-      delegate :deliver_messages, to: :sync_producer, prefix: true
-
       def initialize(config)
         @config = KafkaConfiguration.new(config)
-        @kafka = Kafka.new(**@config.kafka)
       end
 
       def deliver(key:, topic:, payload:, priority:)
         producer(priority).tap do |p|
-          p.produce(payload, key: key, topic: topic)
           case priority
-          when :essential, :standard
-            p.deliver_messages
-          when :batched
-            if p.buffer_size >= batched_message_limit
-              logger.info "Delivering #{p.buffer_size} batched events now"
-              p.deliver_messages
-            end
+          when :essential
+            p.produce_sync(payload: payload, key: key, topic: topic.to_s)
+          when :standard, :low
+            p.produce_async(payload: payload, key: key, topic: topic.to_s)
+          else
+            fail "Unknown priority"
```

Comment:
Would this help us with debugging? 

```suggestion
            fail "Unknown priority #{priority}"
```